test/test_amity.py

```diff
@@ -0,0 +1,415 @@
+        """
+        LivingRoom should raise an InvalidRoomOccupantError exception when Staff tries to join it.
+        """
+        self.staff = staff.Staff("TT", "PP")
+        self.lroom = livingroom.LivingRoom("Livingroom")
+        with self.assertRaises(InvalidRoomOccupantError):
+            self.lroom.add_person(self.staff)
+
+    def test_can_remove_fellow_from_office(self):
+        """
+        Should be able to remove a fellow from office room
+        """
+        self.office = office.Office("Hogwarts")
+        self.f1 = fellow.Fellow("Nan", "Pi")
+        self.f2 = fellow.Fellow("KK", "Brown")
+        self.office.add_person(self.f1)
+        self.office.add_person(self.f2)
+
+        self.assertTupleEqual(
+            (self.office.remove_person(self.f2), self.office.remove_person(self.f1)),
+            (self.f2, self.f1)
+        )
+
+    def test_can_remove_fellows_staff_from_office(self):
+        """
+        Should be able to remove Fellows and Staff from office
+        """
+        self.f1 = fellow.Fellow("Mark", "Mike")
+        self.f2 = fellow.Fellow("JJ", "PP")
+        self.s1 = staff.Staff("FF", "KK")
+        self.s2 = staff.Staff("G", "FF")
+        self.office = office.Office("Small Office")
+        self.office.add_person(self.f1)
+        self.office.add_person(self.f2)
+        self.office.add_person(self.s1)
+        self.office.add_person(self.s2)
+
+        self.assertTupleEqual(
+            (self.office.remove_person(self.s1), self.office.remove_person(self.s2), self.office.remove_person(self.f2),
+             self.office.remove_person(self.f1),),
+            (self.s1, self.s2, self.f2, self.f1)
+        )
+
+    def test_raises_exception_when_removing_non_person_type(self):
+        """
+        Should raise TypeError exception when a non-Person type is removed from the room
+        """
+        self.office = office.Office("Strange Office")
+        self.fellow = fellow.Fellow("Cool", "Guy")
+        with self.assertRaises(TypeError):
+            self.office.remove_person('Cool Guy')
+
+
+class TestAmitySystem(unittest.TestCase):
+    """
+    Tests for the functionality the major components of the system
+    """
+
+    def setUp(self):
+        self.amity = amity.Amity()
+
+    def test_can_add_person_to_list(self):
+        """
+        Should be able to add Person to list of person
+        """
+        self.p1 = fellow.Fellow("New", "Guy")
+        self.p2 = staff.Staff("New", 'Staff')
+        self.amity.add_person(self.p1)
+        self.amity.add_person(self.p2)
+        self.assertTupleEqual(
+            (self.amity.all_persons[0], self.amity.all_persons[1]),
+            (self.p1, self.p2)
+        )
+
+    def test_raises_typeerror_exception_when_adding_non_person_to_list(self):
+        """
+        Should raise a TypeError exception when a non-Person type is added to list
+        """
+        with self.assertRaises(TypeError):
+            self.amity.add_person('Mike Kamau')
```

Comment:
Is this even possible in your code. If you are testing 4 behaviour, there's no way a user will add a non person to a room